### ISSUE TITLE
RTL8192EE causing complete system freeze on Ubuntu 17.10, module insertion doesn't persist after reboot

### ISSUE THREAD

**Issue by peterfreeth** [lwfinger/rtlwifi_new#312]
After Ubuntu updated itself to 17.10 I have had no end of wifi problems. Using an EDIMAX card with a Realtek chip wouldn't work at all, I bought a new TP-Link WN881ND card because it has an Atheros chipset only to find the v2 does not, it uses Realtek again. Realtek support told me flatly that they will not support newer than "2.6.24~4.1" kernel.

I tried the rtlwifi driver, that seemed to work beautifully with very nice instructions, unlike the Realtek official instructions which are wrong. However, after about an hour, my whole system freezes and has to be rebooted, after which point the rtlwifi is gone from the module list and I have to modprobe it again. Yes, I updated initramfs after installing.

Any ideas on what I could try? I'm now on my 3rd wifi card to try and find something that actually works with Ubuntu 17.10.

**Comment by lwfinger** [comment]
Of course module insertion will not last past a reboot. You need to do a 'sudo make install'. That will copy the drivers into the location where the kernel can find them. The only drivers that need to be in initramfs are those needed before the complete environment is established. That is not recommended for ANY wifi driver.

Which branch of this repo are you running? I recommend "extended". The master branch is left here for historical reasons.

Without any logs regarding the freeze, how do you expect any answers?

**Comment by peterfreeth** [comment]
I should have mentioned the obvious, that in order to use the module I had to first build it using the instructions provided, so make then make install, then the instructions further down the page to modprobe.

As to which branch, I simply followed the instructions given here https://github.com/lwfinger/rtlwifi_new so which branch is really dependent on which branch you have set up and given instructions for.

The system freezes, there are no logs as far as I can see because the system is too dead to write any logs. Can you suggest a log file?

**Comment by peterfreeth** [comment]
OK, so instead of following these instructions:

For all distros:
git clone https://github.com/lwfinger/rtlwifi_new.git
cd rtlwifi_new
sudo make install
sudo modprobe -r <<YOUR WIRELESS DRIVER CODE>>
sudo modprobe <<YOUR WIRELESS DRIVER CODE>>

I have downloaded, compiled and installed from the zip file downloaded from the 'extended' option and I will report back on how it goes.

**Comment by peterfreeth** [comment]
After 2 hours my system is still working, and I think it would have crashed by now, so that's good news. The downside is that wifi stopped working after 2 hours. The network still appeared connected but was not passing data. When I turned wireless off and on in the system preferences it immediately reconnected and worked so I will keep an eye on this, as there didn't seem to be any dmesg log entries to indicate what happened.

**Comment by lwfinger** [comment]
As toi which branch are you using, you need to run the command 'git branch'.

**Comment by peterfreeth** [comment]
Thanks

After another hour, I got the disconnect issue with this dmesg output:

[22848.928235] rtlwifi: AP off, try to reconnect now
[22848.928309] wlp2s0: Connection to AP c0:05:c2:ed:5b:d1 lost
[22852.828984] wlp2s0: authenticate with c0:05:c2:ed:5b:d1
[22852.861198] wlp2s0: send auth to c0:05:c2:ed:5b:d1 (try 1/3)
[22852.920021] wlp2s0: authenticated
[22852.923970] wlp2s0: associate with c0:05:c2:ed:5b:d1 (try 1/3)
[22852.935096] wlp2s0: RX AssocResp from c0:05:c2:ed:5b:d1 (capab=0x431 status=0 aid=7)
[22852.945088] wlp2s0: associated
[22959.398631] usblp0: removed
[22977.570610] usblp 1-4:1.1: usblp0: USB Bidirectional printer dev 6 if 1 alt 0 proto 2 vid 0x04A9 pid 0x1763
[23564.595302] rtlwifi: AP off, try to reconnect now
[23564.595339] wlp2s0: Connection to AP c0:05:c2:ed:5b:d1 lost
[23579.964962] IPv6: ADDRCONF(NETDEV_UP): wlp2s0: link is not ready

Turning wifi off & on in network manager and it reconnects.

**Comment by peterfreeth** [comment]
The situation continued as above, no data passed over wifi even though the network was connected, every 1-2 hours, needed wifi to be switched off then back on.

After a reboot, the wifi was again disabled with no adapter found and I had to modprobe rtl8192ee to get it working.

I've now solved this by buying a v1 TP-Link card with the Atheros chipset which works without any configuration required.
